org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/CloseStartTagCodeAction.java

```diff
@@ -0,0 +1,153 @@
+		Range diagnosticRange = diagnostic.getRange();
+		try {
+			int startOffset = document.offsetAt(diagnosticRange.getStart());
+			DOMNode node = document.findNodeAt(startOffset);
+			if (node != null && node.isElement()) {
+				int diagnosticEndOffset = startOffset
+						+ (diagnosticRange.getEnd().getCharacter() - diagnosticRange.getStart().getCharacter());
+				DOMElement element = (DOMElement) node;
+				if (!element.hasStartTag()) {
+					// </foo>
+					DOMElement parent = element.getParentElement();
+					if (parent != null && parent.getTagName() != null) {
+						// <a><b></c>
+						// Replace "c" with "b"
+						String tagName = parent.getTagName();
+						Range replaceRange = XMLPositionUtility.selectEndTagName(element);
+						CodeAction replaceTagAction = CodeActionFactory.replace("Replace with '" + tagName + "' tag",
+								replaceRange, tagName, document.getTextDocument(), diagnostic);
+						codeActions.add(replaceTagAction);
+					}
+				} else {
+					// <foo>
+					boolean startTagClosed = element.isStartTagClosed();
+					char c = document.getText().charAt(diagnosticEndOffset - 1);
+					if (c != '/') {
+						if (startTagClosed) {
+							// ex : <foo attr="" >
+							// // Close with '</$tag>
+							String tagName = element.getTagName();
+							if (tagName != null) {
+								String label = "</" + tagName + ">";
+								String insertText = label;
+								Position endPosition = null;
+								if (!element.hasChildNodes()) {
+									int endOffset = element.getStartTagCloseOffset() + 1;
+									endPosition = document.positionAt(endOffset);
+								} else {
+									String text = document.getText();
+									// the element have some children(Text node, Element node, etc)
+									int endOffset = element.getLastChild().getEnd() - 1;
+									if (endOffset < text.length()) {
+										// remove new line
+										char ch = text.charAt(endOffset);
+										while (ch == '\r' || ch == '\n') {
+											endOffset--;
+											ch = text.charAt(endOffset);
+										}
+									}
+									endOffset++;
+									endPosition = document.positionAt(endOffset);
+									if (hasElements(element)) {
+										// The element have element node as children
+										// the </foo> must be inserted with a new line and indent
+										LineIndentInfo indentInfo = document
+												.getLineIndentInfo(diagnosticRange.getStart().getLine());
+										insertText = indentInfo.getLineDelimiter() + indentInfo.getWhitespacesIndent()
+												+ insertText;
+									}
+								}
+								CodeAction closeEndTagAction = CodeActionFactory.insert("Close with '" + label + "'",
+										endPosition, insertText, document.getTextDocument(), diagnostic);
+								codeActions.add(closeEndTagAction);
+							}
+
+						} else {
+							// ex : <foo attr="
+							// Close with '/>
+							CodeAction autoCloseAction = CodeActionFactory.insert("Close with '/>'",

```

Comment:
I found this case which isn't handled correctly:
```xml
<foo action="toot"
    /
```

![double-backslash-closing-element](https://user-images.githubusercontent.com/22376627/83768752-2d3ebb80-a64d-11ea-912e-eae15b0a2453.gif)

I think this option should be hidden for this case.

Reply:
good catch, fixed